src/VisualStudio/Core/Def/Venus/ContainedLanguage.cs

```diff
@@ -179,5 +165,25 @@ private void OnDataBufferChanged(object sender, TextContentChangedEventArgs e)
             // when primary buffer has changed to update diagnostic positions.
             _diagnosticAnalyzerService.Reanalyze(this.Workspace, documentIds: SpecializedCollections.SingletonEnumerable(this.ContainedDocument.Id));
         }
+
+        public string GetFilePathFromBuffers()
+        {
+            // Try to get the file path from the secondary buffer
+            if (SubjectBuffer.CurrentSnapshot.TextBuffer.Properties.TryGetProperty<ITextDocument>(typeof(ITextDocument), out var secondaryDocument) &&
```

Comment:
>TryGetProperty [](http://example.com/codeflow?start=68&length=14)

Could this use ITextDocumentFactoryService.TryGetTextDocument instead?

Reply:
Given that `ITextDocumentFactoryService` is a MEF part and not all of the users of this contained language are MEF exports themselves (and idek how MEF works in VB), I'm unsure about it.

Reply:
Doesn't the ctor use MEF via stmts like:

_editorAdaptersFactoryService = componentModel.GetService<IVsEditorAdaptersFactoryService>();

Reply:
Ohh I forgot about that. Will try it out.